edi_oca/models/edi_configuration.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2024 Camptocamp SA
+# @author Simone Orsi <simahawk@gmail.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import datetime
+
+import pytz
+
+from odoo import _, api, exceptions, fields, models
+from odoo.tools import DotDict, safe_eval
+
+
+def date_to_datetime(dt):
+    """Convert date to datetime."""
+    if isinstance(dt, datetime.date):
+        return datetime.datetime.combine(dt, datetime.datetime.min.time())
+    return dt
+
+
+def to_utc(dt):
+    """Convert date or datetime to UTC."""
+    # Gracefully convert to datetime if needed 1st
+    return date_to_datetime(dt).astimezone(pytz.UTC)
+
+
+class EdiConfiguration(models.Model):
+    _name = "edi.configuration"
+    _description = """
+        This model is used to configure EDI (Electronic Data Interchange) flows.
+        It allows users to create their own configurations, which can be tailored
+        to meet the specific needs of their business processes.
+    """
+
+    name = fields.Char(string="Name", required=True)
+    active = fields.Boolean(default=True)
+    code = fields.Char(required=True, copy=False, index=True, unique=True)
+    description = fields.Char(help="Describe what the conf is for")
+    backend_id = fields.Many2one(string="Backend", comodel_name="edi.backend")
+    type_id = fields.Many2one(
+        string="Exchange Type",
+        comodel_name="edi.exchange.type",
+        ondelete="cascade",
+        auto_join=True,
+        index=True,
+    )
+    model = fields.Many2one(
+        "ir.model",
+        string="Model",
+        help="Model the conf applies to. Leave blank to apply for all models",
+    )
+    model_name = fields.Char(related="model.model", store=True)
+    trigger = fields.Selection(
+        [
+            ("on_record_write", "Update Record"),
+            ("on_record_create", "Create Record"),
+            ("on_email_send", "Send Email"),
```

Comment:
I guess this is to ease the configuration for models like SO, PO, INV, that can be sent by email.
As there's no specific implementation here, I would add a comment that explain why this is here.

Also, I think `trigger` should be mandatory otherwise the conf is useless, no?
Maybe we can add another option like `("disabled", "Disabled")` which would allow to keep the conf visible but disabled.